### PR TITLE
Adds support for class completions after ASI inserted class property definition

### LINKED ISSUE #30536: No member suggestions after class property definition without  semicolon
From https://github.com/Microsoft/vscode/issues/69897

**TypeScript Version:**  3.4.0-rc

**Search Terms:**
- suggestion, suggest, completions

**Code**
For the javascript

```js
// @ts-check

import React from 'react';

class X extends React.Component {
    x = 1
    comp
}
```

1. Trigger suggestions after `comp`

**Expected behavior:**
Suggestions for the react `componentDid...` methods returned

**Actual behavior:**
Global suggestions returned

If you add a semicolon after `x = 1;`, you get the correct suggestions:

```ts
// @ts-check

import React from 'react';

class X extends React.Component {
    x = 1
    comp
}
```
**Playground Link:** 

**Related Issues:**

### PR DESCRIPTION
**Pull request by orta**
Fixes #30536 - handles both the case where you start writing on a newline directly, and when you are continuing to write inside an identifier.

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -2179,7 +2185,13 @@ namespace ts.Completions {
             case SyntaxKind.CommaToken: // class c {getValue(): number, | }
                 return tryCast(contextToken.parent, isObjectTypeDeclaration);
             default:
-                if (!isFromObjectTypeDeclaration(contextToken)) return undefined;
+                if (!isFromObjectTypeDeclaration(contextToken)) {
+                    // class c extends React.Component { a: () => 1\n| }
+                    if (getLineAndCharacterOfPosition(sourceFile, contextToken.getEnd()).line !== getLineAndCharacterOfPosition(sourceFile, position).line && isObjectTypeDeclaration(location)) {
```
Comment (andrewbranch) [inline src/services/completions.ts]: Adding the `isObjectTypeDeclaration(location)` miiiight have been overly conservative, but I suggested it to err on the side of caution—the smallest reasonable net I could think of that would catch the repros provided.

Comment (sheetalkamat) [inline src/services/completions.ts]: Can you please reverse the order of checks here (that is checking `isObjectTypeDeclaration(location)` before line?

Comment (orta) [reply src/services/completions.ts]: Switching the if ordering changes the behavior from what it previously was, and starts to trigger behavior which I think we don't want.

[This fourslash](https://github.com/Microsoft/TypeScript/blob/e55f97ec282d720f9bf58792ece0acc048202f93/tests/cases/fourslash/completionsGeneratorFunctions.ts)  test fails:
```
/// <reference path="fourslash.ts" />

////function /*a*/ ;
////function* /*b*/ ;
////interface I {
////    abstract baseMethod(): Iterable<number>;
////}
////class C implements I {
////    */*c*/ ;
////    public */*d*/
////}
////const o: I = {
////    */*e*/
////};
////1 * /*f*/

verify.completions(
    { marker: ["a", "b"], exact: undefined, isNewIdentifierLocation: true },
    { marker: ["c", "d"], exact: ["baseMethod"], isNewIdentifierLocation: true },
    { marker: "e", exact: ["baseMethod"] },
    { marker: "f", includes: [{ name: "Number", sortText: completion.SortText.GlobalsOrKeywords }] },
);
```

With 

```
  1) fourslash tests
       tests/cases/fourslash/completionsGeneratorFunctions.ts
         fourslash test completionsGeneratorFunctions.ts runs correctly:
     Error: At f: Expected 'isNewIdentifierLocation' to be false, got true
```

Which is returning an empty array of completions

**Comment by orta** [comment]
Alright I'm merging this 👍

**Pull request merged**